### PR TITLE
Chore/api reference bucket schemas

### PR DESCRIPTION
**Pull request by sunbryely-influxdata**
- List Bucket Schemas
- Create Bucket Schemas
- Retrieve Bucket Schemas
- Update Bucket Schemas

### CODE REVIEW

File: src/cloud/paths/measurements.yml [hunk 1]
```diff
@@ -38,26 +53,64 @@ get:
...
+
+    #### Limitations
+
+    - Buckets must be created with an "explict" `schemaType` in order to use
+    schemas.
+
+    #### Related guides
+
+    - [Manage bucket schemas]({{% INFLUXDB_DOCS_URL %}}/organizations/buckets/bucket-schema/).
+    - [Using bucket schemas](https://www.influxdata.com/blog/new-bucket-schema-option-protect-from-unwanted-schema-changes/)
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:93]:
```suggestion
    - [Create a bucket with an explicit schema]({% INFLUXDB_DOCS_URL %}}/organizations/buckets/create-bucket/#create-a-bucket-with-an-explicit-schema)
```

File: src/cloud/paths/measurements.yml [hunk 2]
```diff
@@ -1,26 +1,41 @@
 summary: Bucket schemas
 get:
   summary: List all measurement schemas of a bucket
```
[the remark this answers is not in the document]
Comment (jstirnaman) [reply src/cloud/paths/measurements.yml:3]: Looks like a need to expand the guide to using the API for bucket schemas?

File: src/cloud/paths/measurements.yml [hunk 3]
```diff
@@ -1,26 +1,41 @@
...
+    - [Using bucket schemas](https://www.influxdata.com/blog/new-bucket-schema-option-protect-from-unwanted-schema-changes/)
+
   operationId: getMeasurementSchemas
   parameters:
     - in: query
       name: org
-      description: The name of the organization.
+      description: |
+        An organization name.
+        Only returns schemas owned by this organization.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:18]:
```suggestion
        Only returns schemas owned by the specified organization.
```
a la Kelly

File: src/cloud/paths/measurements.yml [hunk 4]
```diff
@@ -1,26 +1,41 @@
...
       schema:
         type: string
     - in: query
       name: orgID
-      description: The identifier of the organization.
+      description: |
+        An organization ID.
+        Only returns schemas owned by this organization.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:25]:
```suggestion
        Only returns schemas owned by the specified organization.
```

File: src/cloud/paths/measurements.yml [hunk 5]
```diff
@@ -1,26 +1,41 @@
...
       schema:
         type: string
     - in: query
       name: name
-      description: The name of the measurement to return schemas for.
+      description: |
+        A measurement name.
+        Only returns measurement schemas with this name.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:32]:
```suggestion
        Only returns measurement schemas with the specified name.
```

File: src/cloud/paths/measurements.yml [hunk 6]
```diff
@@ -1,26 +1,41 @@
...
       schema:
         type: string
     - in: path
       name: bucketID
-      description: The identifier of the bucket.
+      description: |
+        The ID of the bucket to retrieve schemas from.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:38]:
```suggestion
        A bucket ID.
        Lists measurement schemas for the specified bucket.
```

File: src/cloud/paths/measurements.yml [hunk 7]
```diff
@@ -1,26 +1,41 @@
 summary: Bucket schemas
 get:
   summary: List all measurement schemas of a bucket
+  description: |
+    Retrieves a list of measurement [schemas]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#schema)
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:5]:
```suggestion
    Retrieves a list of explicitly defined measurement [schemas]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#schema)
```

Explicit only?

File: src/cloud/paths/measurements.yml [hunk 8]
```diff
@@ -38,26 +53,64 @@ get:
...
     '404':
-      description: Bucket not found
-      $ref: '../../common/responses/ServerError.yml'
+      $ref: "../../common/responses/ResourceNotFoundError.yml"
+    "500":
+      $ref: "../../common/responses/InternalServerError.yml"
 post:
   summary: Create a measurement schema for a bucket
+  description: |
+    Creates a measurement [schema]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#schema)
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:79]:
```suggestion
    Creates an explicit measurement [schema]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#schema)
```

File: src/cloud/paths/measurements.yml [hunk 9]
```diff
@@ -38,26 +53,64 @@ get:
...
+    - [Manage bucket schemas]({{% INFLUXDB_DOCS_URL %}}/organizations/buckets/bucket-schema/).
+    - [Using bucket schemas](https://www.influxdata.com/blog/new-bucket-schema-option-protect-from-unwanted-schema-changes/)
   operationId: createMeasurementSchema
   parameters:
     - in: query
       name: org
-      description: The name of the organization.
+      description: |
+        An organization name.
+        Only returns schemas owned by this organization.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:100]:
```suggestion
        Only returns schemas owned by the specified organization.
```

File: src/cloud/paths/measurements.yml [hunk 10]
```diff
@@ -38,26 +53,64 @@ get:
...
       schema:
         type: string
     - in: query
       name: orgID
-      description: The identifier of the organization.
+      description: |
+        An organization ID.
+        Only returns schemas owned by this organization.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:107]:
```suggestion
        Only returns schemas owned by the specified organization.
```

File: src/cloud/paths/measurements.yml [hunk 11]
```diff
@@ -38,26 +53,64 @@ get:
...
       schema:
         type: string
     - in: path
       name: bucketID
-      description: The identifier of the bucket.
+      description: |
+        The ID of the bucket to add a schema to.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:113]:
```suggestion
        A bucket ID.
        Adds a schema for the specified bucket.
```

File: src/cloud/paths/measurements_measurementID.yml [hunk 12]
```diff
@@ -1,27 +1,35 @@
 summary: Bucket Schema
 get:
-  summary: Retrieve measurement schema information
+  summary: Retrieve a measurement schema
+  description: |
+    Retrieves a measurement [schemas]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#schema).
```
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:5]:
```suggestion
    Retrieves a measurement [schema]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#schema).
```

File: src/cloud/paths/measurements_measurementID.yml [hunk 13]
```diff
@@ -1,27 +1,35 @@
...
   operationId: getMeasurementSchema
   parameters:
     - in: query
       name: org
-      description: The name of the organization.
+      description: |
+        Organization name.
+        Only returns schemas owned by this organization.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:12]:
```suggestion
        Only returns schemas owned by the specified organization.
```

File: src/cloud/paths/measurements_measurementID.yml [hunk 14]
```diff
@@ -1,27 +1,35 @@
...
       schema:
         type: string
     - in: query
       name: orgID
-      description: The identifier of the organization.
+      description: |
+        Organization ID.
+        Only returns schemas owned by this organization.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:19]:
```suggestion
        Only returns schemas owned by the specified organization.
```

File: src/cloud/paths/measurements_measurementID.yml [hunk 15]
```diff
@@ -1,27 +1,35 @@
...
       schema:
         type: string
     - in: path
       name: bucketID
-      description: The identifier of the bucket.
+      description: |
+        The ID of the bucket to retrieve schemas from.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:25]:
```suggestion
        A bucket ID.
        Retrieves schemas for the specified bucket.
```

File: src/cloud/paths/measurements_measurementID.yml (hunk 12 above)
Comment (jstirnaman) [reply src/cloud/paths/measurements_measurementID.yml:5]: "explicit"?

File: src/cloud/paths/measurements_measurementID.yml [hunk 16]
```diff
@@ -1,27 +1,35 @@
...
       schema:
         type: string
       required: true
     - in: path
       name: measurementID
-      description: The identifier of the measurement.
+      description: |
+        The ID of the measurement schema to retrieve.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:32]:
```suggestion
        The measurement schema ID.
        Specifies the measurement schema to retrieve.
```

File: src/cloud/paths/measurements_measurementID.yml [hunk 17]
```diff
@@ -35,34 +43,59 @@ get:
...
+      $ref: "../../common/responses/ResourceNotFoundError.yml"
+    "500":
+      $ref: "../../common/responses/InternalServerError.yml"
 patch:
   summary: Update a measurement schema
+  description: |
+    Updates a measurement [schema]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#schema).
+
+    Use this endpoint to update the columns of a measurement schema. The `name`,
+    `type`, and `dataType` fields of the column can be updated.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:62]:
```suggestion
    Use this endpoint to update the fields (`name`, `type`, and `dataType`) of measurement schema columns . 
```

File: src/cloud/paths/measurements_measurementID.yml [hunk 18]
```diff
@@ -35,34 +43,59 @@ get:
...
+
+    #### Limitations
+
+    - The `name` of the measurement can't be updated.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:66]:
```suggestion
    - You can't update the `name` of a measurement.
```

File: src/cloud/paths/measurements_measurementID.yml [hunk 19]
```diff
@@ -35,34 +43,59 @@ get:
...
+    - [Manage bucket schemas]({{% INFLUXDB_DOCS_URL %}}/organizations/buckets/bucket-schema/).
+    - [Using bucket schemas](https://www.influxdata.com/blog/new-bucket-schema-option-protect-from-unwanted-schema-changes/).
   operationId: updateMeasurementSchema
   parameters:
     - in: query
       name: org
-      description: The name of the organization.
+      description: |
+        Organization name.
+        Only returns schemas owned by this organization.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:78]:
```suggestion
        Only returns schemas owned by the specified organization.
```

File: src/cloud/paths/measurements_measurementID.yml [hunk 20]
```diff
@@ -35,34 +43,59 @@ get:
...
       schema:
         type: string
     - in: query
       name: orgID
-      description: The identifier of the organization.
+      description: |
+        Organization ID.
+        Only returns schemas owned by this organization.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:85]:
```suggestion
        Only returns schemas owned by the specified organization.
```

File: src/cloud/paths/measurements_measurementID.yml [hunk 21]
```diff
@@ -35,34 +43,59 @@ get:
...
+      description: |
+        Organization name.
+        Only returns schemas owned by this organization.
       schema:
         type: string
     - in: query
       name: orgID
-      description: The identifier of the organization.
+      description: |
+        Organization ID.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:84]:
```suggestion
        An organization ID.
```

File: src/cloud/paths/measurements_measurementID.yml [hunk 22]
```diff
@@ -35,34 +43,59 @@ get:
...
+
+    - [Manage bucket schemas]({{% INFLUXDB_DOCS_URL %}}/organizations/buckets/bucket-schema/).
+    - [Using bucket schemas](https://www.influxdata.com/blog/new-bucket-schema-option-protect-from-unwanted-schema-changes/).
   operationId: updateMeasurementSchema
   parameters:
     - in: query
       name: org
-      description: The name of the organization.
+      description: |
+        Organization name.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:77]:
```suggestion
        An organization name.
```

File: src/cloud/paths/measurements_measurementID.yml (hunk 21 above)
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:84]:
```suggestion
        An organization ID.
```

File: src/cloud/paths/measurements_measurementID.yml [hunk 23]
```diff
@@ -35,34 +43,59 @@ get:
...
+      description: |
+        Organization ID.
+        Only returns schemas owned by this organization.
       schema:
         type: string
     - in: path
       name: bucketID
-      description: The identifier of the bucket.
+      description: |
+        The ID of the bucket to retrieve schemas from.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:91]:
```suggestion
        A bucket ID.
        Specifies the bucket to retrieve schemas for.
```

File: src/cloud/paths/measurements_measurementID.yml [hunk 24]
```diff
@@ -35,34 +43,59 @@ get:
...
       schema:
         type: string
       required: true
     - in: path
       name: measurementID
-      description: The identifier of the measurement.
+      description: |
+        The ID of the measurement schema to retrieve.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:98]:
```suggestion
        A measurement schema ID.
        Retrieves the specified measurement schema.
```

File: src/cloud/schemas/MeasurementSchemaCreateRequest.yml [hunk 25]
```diff
@@ -21,6 +21,8 @@
     - columns
   properties:
     name:
+      description: |
+        The name of the measurement.
```
Comment (jstirnaman) [inline src/cloud/schemas/MeasurementSchemaCreateRequest.yml:25]:
```suggestion
        The measurement name.
```

**Review by jstirnaman — CHANGES_REQUESTED** [review]

File: src/cloud/paths/measurements.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (sunbryely-influxdata) [reply src/cloud/paths/measurements.yml]: @jstirnaman To clarify, are you asking for a guide within this description, or an external guide?

File: src/cloud/paths/measurements.yml [hunk 26]
```diff
@@ -1,26 +1,42 @@
 summary: Bucket schemas
 get:
-  summary: List all measurement schemas of a bucket
+  summary: List measurement schemas of a bucket
+  description: |
+    Retrieves a list of explicitly defined measurement
+    [schemas]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#schema) for a bucket.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:6]:
```suggestion
    [schemas]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#schema) (`"schemaType": "explicit"`) for a bucket.
```

Comment (jstirnaman) [reply src/cloud/paths/measurements.yml:6]: The type value is `explicit`, so we should just say _explicit_ bucket schemas and link to a definition or explain "explicitly defined..."

File: src/cloud/paths/measurements.yml [hunk 27]
```diff
@@ -38,26 +54,65 @@ get:
...
     '404':
-      description: Bucket not found
-      $ref: '../../common/responses/ServerError.yml'
+      $ref: "../../common/responses/ResourceNotFoundError.yml"
+    "500":
+      $ref: "../../common/responses/InternalServerError.yml"
 post:
   summary: Create a measurement schema for a bucket
+  description: |
+    Creates an explict measurement [schema]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#schema)
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:80]:
```suggestion
    Creates an _explict_ measurement [schema]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#schema)
```

File: src/cloud/paths/measurements.yml [hunk 28]
```diff
@@ -70,7 +125,9 @@ post:
           $ref: "../schemas/MeasurementSchemaCreateRequest.yml"
   responses:
     '201':
-      description: The created measurement schema.
+      description: |
+        Success.
+        The response body contains the measurement schema.
       headers:
         ETag:
           description: The current version of the measurement schema
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:133]:
```suggestion
          description: The current version of the measurement schema.
```

File: src/cloud/paths/measurements.yml [hunk 29]
```diff
@@ -38,26 +54,65 @@ get:
...
+    - [Manage bucket schemas]({{% INFLUXDB_DOCS_URL %}}/organizations/buckets/bucket-schema/).
+    - [Create a bucket with an explicit schema]({% INFLUXDB_DOCS_URL %}}/organizations/buckets/create-bucket/#create-a-bucket-with-an-explicit-schema)
   operationId: createMeasurementSchema
   parameters:
     - in: query
       name: org
-      description: The name of the organization.
+      description: |
+        An organization name.
+        Only returns schemas owned by the specified organization.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:101]:
```suggestion
        Specifies the organization that owns the schema.
```

File: src/cloud/paths/measurements.yml [hunk 30]
```diff
@@ -38,26 +54,65 @@ get:
...
       schema:
         type: string
     - in: query
       name: orgID
-      description: The identifier of the organization.
+      description: |
+        An organization ID.
+        Only returns schemas owned by the specified organization.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:108]:
```suggestion
        Specifies the organization that owns the schema.
```

File: src/cloud/paths/measurements_measurementID.yml [hunk 31]
```diff
@@ -35,34 +45,61 @@ get:
...
+    - [Manage bucket schemas]({{% INFLUXDB_DOCS_URL %}}/organizations/buckets/bucket-schema/).
+    - [Using bucket schemas](https://www.influxdata.com/blog/new-bucket-schema-option-protect-from-unwanted-schema-changes/).
   operationId: updateMeasurementSchema
   parameters:
     - in: query
       name: org
-      description: The name of the organization.
+      description: |
+        An organization name.
+        Only returns schemas owned by the specified organization.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:80]:
```suggestion
        Specifies the organization that owns the schema.
```

File: src/cloud/paths/measurements_measurementID.yml [hunk 32]
```diff
@@ -35,34 +45,61 @@ get:
...
       schema:
         type: string
     - in: query
       name: orgID
-      description: The identifier of the organization.
+      description: |
+        An organization ID.
+        Only returns schemas owned by the specified organization.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:87]:
```suggestion
        Specifies the organization that owns the schema.
```

File: src/cloud/paths/measurements_measurementID.yml [hunk 33]
```diff
@@ -35,34 +45,61 @@ get:
...
+        Specifies the bucket to retrieve schemas for.
       schema:
         type: string
       required: true
     - in: path
       name: measurementID
-      description: The identifier of the measurement.
+      description: |
+        A measurement schema ID.
+        Retrieves the specified measurement schema.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements_measurementID.yml:102]:
```suggestion
        Updates the specified measurement schema.
```

File: src/cloud/schemas/MeasurementSchemaCreateRequest.yml [hunk 34]
```diff
@@ -21,6 +21,8 @@
     - columns
   properties:
     name:
+      description: |
+        The measurement name.
```
Comment (jstirnaman) [inline src/cloud/schemas/MeasurementSchemaCreateRequest.yml:25]: Add a link (glossary, etc.) for measurement or measurement name.

**Review by jstirnaman — CHANGES_REQUESTED** [review]

File: src/cloud/paths/measurements.yml [hunk 35]
```diff
@@ -1,26 +1,49 @@
...
+  summary: List measurement schemas of a bucket
+  description: |
+    Retrieves a list of _explicit_
+    [schemas]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#schema)
+    (`"schemaType": "explicit"`) for a bucket.
+
+    _Explicit_ schemas are used to enforce column names, tags, fields, and data
+    types for your data.
+
+    By default, buckets have _implicit_ schema-type (`"schemaType": "implicit"`)
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:12]:
```suggestion
    By default, buckets have an _implicit_ schema-type (`"schemaType": "implicit"`)
```

File: src/cloud/paths/measurements.yml [hunk 36]
```diff
@@ -1,26 +1,49 @@
...
+    that conform to your data.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:13]:
```suggestion
    that conforms to your data.
```

File: src/cloud/paths/measurements.yml (hunk 35 above)
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:12]: 👍

File: src/cloud/paths/measurements.yml [hunk 37]
```diff
@@ -38,26 +61,70 @@ get:
...
   summary: Create a measurement schema for a bucket
+  description: |
+    Creates an _explict_ measurement [schema]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#schema)
+    for a bucket.
+
+    _Explicit_ schemas are used to enforce column names, tags, fields, and data
+    types for your data.
+
+    By default, buckets have _implicit_ schema-type (`"schemaType": "implicit"`)
+    that conform to your data.
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:94]:
```suggestion
    By default, buckets have an _implicit_ schema-type (`"schemaType": "implicit"`)
    that conforms to your data.
```

File: src/cloud/paths/measurements.yml [hunk 38]
```diff
@@ -38,26 +61,70 @@ get:
...
+
+    Use this endpoint to create schemas that prevent non-conforming write requests.
+
+    #### Limitations
+
+    - Buckets must be created with an "explict" `schemaType` in order to use
```
Comment (jstirnaman) [inline src/cloud/paths/measurements.yml:100]:
```suggestion
    - Buckets must be created with the "explict" `schemaType` in order to use
```

**Review by jstirnaman — APPROVED** [review]
A few minor fixes.

**Pull request merged**